go-controller/pkg/ovn/ovn.go

```diff
@@ -592,11 +616,34 @@ func (oc *Controller) WatchPods() {
 				}
 				// deletion was a success; remove delete retry entry
 				oc.removeDeleteRetry(pod)
+			} else if util.PodCompleted(pod) {
```

Comment:
should we put this chunk under this if else as a separate function and call it from add and update handlers? it might reduce redundancy? or do you prefer to keep this here? I'm 50-50 :D

Reply:
yeah I thought about that too, but in the end I decided I wanted the clarity of reading inside the handler adds.